vortex-array/src/array/sparse/mod.rs

```diff
@@ -20,11 +20,10 @@ impl_encoding!("vortex.sparse", ids::SPARSE, Sparse);
 
 #[derive(Debug, Clone, Serialize, Deserialize)]
 pub struct SparseMetadata {
-    indices_dtype: DType,
+    indices_ptype: PType,
     // Offset value for patch indices as a result of slicing
     indices_offset: usize,
     indices_len: usize,
-    len: usize,
     fill_value: Scalar,
```

Comment:
this can be `ScalarValue` as well, the fill has to be of `self.dtype()`

Reply:
That required around +269/-225 of changes so I put it into a follow up PR: https://github.com/spiraldb/vortex/pull/955